crates/miden-tx/src/executor/mod.rs

```diff
@@ -143,7 +144,7 @@ where
         block_ref: BlockNumber,
         notes: InputNotes<InputNote>,
         tx_args: TransactionArgs,
-        source_manager: Arc<dyn SourceManager + Send + Sync>,
+        source_manager: Arc<dyn SourceManagerSync>,
```

Comment:
Could we make `source_manager` a construction-time parameter? Conceptually, it is in the same category as to the data store and the authenticator.